### ISSUE TITLE
[0.62] Cannot read property 'sourceDir' of null

### ISSUE THREAD

**Issue by olcayertas** [microsoft/react-native-windows#5752]
I am getting error when installing pods.

## Environment
1. `npx react-native --version`: 4.12.0
2. `npx react-native run-windows --info`: 
```
System:
    OS: macOS 10.15.6
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 902.91 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.1.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  npmPackages:
    @react-native-community/cli: 4.10.1 => 4.10.1 
    react: 16.13.1 => 16.13.1 
    react-native: 0.62.2 => 0.62.2 
    react-native-windows: 0.62.0 => 0.62.0
  Installed UWP SDKs:
```

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Add some library for native
2. Run `pod install`

## Expected Results
Expected to install without error.

Instead getting this error:

```
[!] Invalid `Podfile` file: 767: unexpected token at 'TypeError: Cannot read property 'sourceDir' of null

    at Object.dependencyConfigWindows [as dependencyConfig] (/Users/myusername/Projects/myproject/node_modules/react-native-windows/local-cli/config/dependencyConfig.js:7:32)

    at /Users/myusername/Projects/myproject/node_modules/@react-native-community/cli/build/tools/config/index.js:63:61

    at Array.reduce (<anonymous>)

    at getDependencyConfig (/Users/myusername/Projects/myproject/node_modules/@react-native-community/cli/build/tools/config/index.js:60:51)

    at Object.get react-native-safe-area-context [as react-native-safe-area-context] (/Users/myusername/Projects/myproject/node_modules/@react-native-community/cli/build/tools/config/index.js:135:18)

    at /Users/myusername/Projects/myproject/node_modules/@react-native-community/cli/build/commands/config/config.js:16:51

    at Array.forEach (<anonymous>)

    at filterConfig (/Users/myusername/Projects/myproject/node_modules/@react-native-community/cli/build/commands/config/config.js:15:38)

    at Object.func (/Users/myusername/Projects/myproject/node_modules/@react-native-community/cli/build/commands/config/config.js:27:32)

    at Command.handleAction (/Users/myusername/Projects/myproject/node_modules/@react-native-community/cli/build/index.js:186:23)
'.

 #  from /Users/myusername/Projects/myproject/ios/Podfile:49
 #  -------------------------------------------
 #  
 >    use_native_modules!
 #  
 #  -------------------------------------------
```

**Quick Fix:**

Change the line 7 in `local-cli/config/dependencyConfig.js`
```
function dependencyConfigWindows(folder, userConfig = {}) {
  const sourceDir = userConfig.sourceDir || findWindowsAppFolder(folder);
```
with this:
```
function dependencyConfigWindows(folder, userConfig = {}) {
  const sourceDir = userConfig & userConfig.sourceDir || findWindowsAppFolder(folder);
```

**Comment by NickGerleman** [comment]
Thanks for finding this! Just to clarify the issue, is the scenario using a project supporting both macOS and Windows on macOS?

Not sure we've had strong consideration for cases where hooks to the Windows CLI are loaded on other platforms. We do ideally want as much of our CLI code to be xplat as possible, and this is a really good example of why that matters. FYI @jonthysell who added config code and will have more context on this specific issue.

I've been wanting to spin up some macOS tests for some of the xplat JS in our repo. Wondering if we want to test CLI code as well, or consider ways to gracefully bail early where we don't have tests (not sure we do for config yet).

**Comment by jonthysell** [comment]
All of this is working as intended in 63. This is a bug in 62, which had an early attempt at config/autolinking before I implemented in end-to-end in 63.

**Comment by jonthysell** [comment]
In fact, since what appears in 62 for config/autolinking is incomplete, untested, and not compatible with what we did in 63, I think we should stub out the code in 62.

**Comment by NickGerleman** [comment]
This should be fixed in 0.62.6 which will be published Monday morning.
